### PR TITLE
[Feature] FA2 support

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description
This PR adds FA2 support:
- Adding an FA2 field in the new proposal page
- Handling FA2 in history and proposals

### CODE REVIEW

**Review by rueshyna — DISMISSED** [review]
LGTM. (I push some small change)

File: components/ProposalCard.tsx [hunk 1]
```diff
@@ -80,13 +80,45 @@ export const RenderProposalContent = ({
...
+          fa2_address: contractData.payload.fa2_address,
+          token_id: contractData.payload.token_id.toString(),
+        }),
+      };
+    } else if (metadata.entrypoint === "%transfer") {
+      const [
+        {
+          txs: [{ to_, token_id, amount }],
+        },
+      ] = metadata.payload;
```
[the remark this answers is not in the document]
Comment (rueshyna) [reply components/ProposalCard.tsx:102]: This haven't fixed yet. :) 
`KT19ojdAYP9B9ZPEtoRYkj6o6eg7UEEGBkN7` import this address and go proposals page. 
You'll see what I mean

[the remark this answers is not in the document]
Comment (rueshyna) [reply components/ProposalCard.tsx:102]: `KT1Pon4g1YtudAQg2rRUkET4PxjgBKyEEn4S` sorry, should import this one

[the remark this answers is not in the document]
Comment (JulesGuesnon) [reply components/ProposalCard.tsx:107]: Yeah you're right, my bad, my check was pretty bad 😓 It's fixed now

File: components/ProposalCard.tsx [hunk 2]
```diff
@@ -94,7 +94,12 @@ export const RenderProposalContent = ({
           token_id: contractData.payload.token_id.toString(),
         }),
       };
-    } else if (metadata.entrypoint === "%transfer" && !!metadata.payload) {
+    } else if (
+      metadata.entrypoint === "%transfer" &&
+      Array.isArray(metadata.payload) &&
+      !!metadata.payload[0]?.txs
+    ) {
+      console.log(metadata.payload);
```
Comment (rueshyna) [inline components/ProposalCard.tsx:102]: console.log!

File: components/ProposalCard.tsx [hunk 3]
```diff
@@ -94,7 +94,12 @@ export const RenderProposalContent = ({
           token_id: contractData.payload.token_id.toString(),
         }),
       };
-    } else if (metadata.entrypoint === "%transfer" && !!metadata.payload) {
+    } else if (
+      metadata.entrypoint === "%transfer" &&
+      Array.isArray(metadata.payload) &&
+      !!metadata.payload[0]?.txs
```
Comment (rueshyna) [inline components/ProposalCard.tsx:100]: I'd like also check if `txs` exist.
- txs.to_
- txs.token_id
- txs.amount

**Review by rueshyna — APPROVED** [review]
thanks for the work :100:

**Pull request merged**